torch/distributed/utils.py

```diff
@@ -16,20 +18,26 @@ def _recursive_to(inputs, target_gpu, use_side_stream_for_tensor_copies):
     """
 
     def to_map(obj):
+        ddp_device_type = _get_available_device_type()
```

Comment:
so this will return XPU when XPU is available and CUDA when CUDA is available? what happens when both are available, is there a priority assigned and can we override this?

Reply:
> so this will return XPU when XPU is available and CUDA when CUDA is available? what happens when both are available, is there a priority assigned and can we override this?

@rohan-varma Thanks for your reply. The _get_available_device_type already gives the priority, cuda will be the highest priority, and cuda and xpu cannot alive together. Any suggestions for this?